### PR TITLE
manage Ivits and Xenos during setup

### PR DESCRIPTION
**Pull request by rzulian**

### CODE REVIEW

File: src/engine.spec.ts [hunk 1]
```diff
@@ -3,6 +3,29 @@ import Engine from "..";
...
+
+  it("should throw when trying to build on the wrong place", () => {
+    const moves = [
+      "init 2 randomSeed",
+      "p1 faction terrans",
+      "p2 faction xenos",
+      "p1 build m 0x0",
+    ];
+
+    expect(() => new Engine(moves)).to.not.throw();
```
Comment (coyotte508) [inline src/engine.spec.ts]: It should be `expect(() => new Engine(moves)).to.throw();`

File: src/engine.spec.ts [hunk 2]
```diff
@@ -3,6 +3,29 @@ import Engine from "..";
...
+  });
+
+  it("should process a simple game with Xenos (three mines) without errors", () => {
+    const moves = [
+      "init 2 randomSeed",
+      "p1 faction terrans",
+      "p2 faction xenos",
+      "p1 build m 2x2",
+      "p2 build m -2x5"
```
Comment (coyotte508) [inline src/engine.spec.ts]: There are only one mines for each, shouldn't p2 build three mines?

File: src/engine.ts [hunk 3]
```diff
@@ -57,7 +58,21 @@ export default class Engine {
...
+    player = this.players.findIndex(pl =>  pl.faction === Faction.Xenos) ;
+
+    if (player !== -1) {
+      return player;
+    }
+
+     // Ivits is last 
+     player = this.players.findIndex(pl =>  pl.faction === Faction.Ivits) ;
+
+     if (player !== -1) {
```
Comment (coyotte508) [inline src/engine.ts]: Should also check that they haven't built their PI

**Comment by rzulian** [comment]
added control for Ivits

**Pull request merged**